azure/application-gateway/advanced/main.tf

```diff
@@ -328,9 +328,5 @@ resource "azurerm_application_gateway" "main" {
       error_message = "For every backend_http_setting, both the backend_http_settings https_enabled option and the related probe https_enabled option must be the same value."
       condition     = alltrue([for k, v in var.backend_http_settings : var.probe[v.probe_name].https_enabled == v.https_enabled])
     }
-    precondition {
```

Comment:
Why does this need to be removed

Reply:
@jk-ans because there was no point in having a `waf_enabled` function if it was going to be directly tied to `waf_configuration`. By setting `waf_configuration` we can imply that it's `waf_enabled`.